Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -791,6 +791,10 @@ lemma exists_iff_succAbove {P : Fin (n + 1) → Prop} (p : Fin (n + 1)) :
     · exact .inr ⟨_, hi⟩
   mpr := by rintro (h | ⟨i, hi⟩) <;> exact ⟨_, ‹_›⟩
 
+/-- Analogue of `Fin.eq_zero_or_eq_succ` for `succAbove`. -/
+theorem eq_self_or_eq_succAbove (i j : Fin (n + 1)) : i = j ∨ (∃ k, i = j.succAbove k) :=
```

Comment:
Please always try calling `p` the first argument to `succAbove`, as its the **p**ivot and doesn't play the same role at all as the second argument.
```suggestion
theorem eq_self_or_eq_succAbove (p i : Fin (n + 1)) : i = p ∨ ∃ j, i = p.succAbove j :=
```
Here and everywhere else